### ISSUE TITLE
Question: How to use shazamio for big lists (e.g. >500)?

### ISSUE THREAD

**Issue by asapsmc** [shazamio/ShazamIO#120]
Is there any recommended approach for when we need to retrieve a big list of shazams? I'm working with a list of more than 500 files, and I'm using the same code as in one of your examples: 
```
Shazam(
        http_client=HTTPClient(
            retry_options=ExponentialRetry(
                attempts=12, max_timeout=204.8, statuses={500, 502, 503, 504, 429}
            ),
        ),
```
I've also implemented a sort of manual shazam_cache, to avoid repeating the same audio Shazam. 
However, I get many errors. Firstly, I start getting '429' errors, but in reattempts, I start getting  mostly "Failed to decode json" errors. 
Could anyone recommend an approach to handle these errors more effectively, especially when dealing with large lists? Any tips on optimizing the retry logic or managing rate limits would be greatly appreciated.

Thanks in advance for your help!

**Comment by wttdotm** [comment]
yo @asapsmc did you figure this out? i have a backlog of several hundreds of gigs of songs I need to identify lol

**Comment by asapsmc** [comment]
No, I couldn't figure any smart way to solve this. If you do, please share it.

**Comment by wttdotm** [comment]
@asapsmc VPNing into a different IP address works. 

I'm using Nord, not sure if it's like rotating IPs in the background or something, but I was able to start my script, go to sleep, and get through like 10k songs overnight so something must be working.

**Comment by AlKun25** [comment]
I am trying to create an open-source app that might use this package. I have similar kind of an issue, where I want to decide if I can host the application in case the number of calls are going to be too many (greater than API limits). 

Is this package tracking the API calls based on the IP address, given that I have not shared any API key with it?

**Comment by wttdotm** [comment]
I doubt this package is tracking anything, my understanding is just an easily usable way to make the Shazam API calls. Whether or not Shazam's API is tracking who is sending them requests is a different matter, and it is indeed tracking them given that we are here in a thread talking about rate limiting. If you want to implement some logic to avoid running into that problem, that's something you'll probably just have to build on top of this, tracking your calls you make with this package and then doing whatever IP swap you want or other technique you have whenever you hit some X requests threshold.
